src/hotspot/share/oops/oop.hpp

```diff
@@ -77,6 +77,7 @@ class oopDesc {
   inline Klass* klass_or_null() const;
   inline Klass* klass_or_null_acquire() const;
 
+  narrowKlass narrow_klass() const { return _metadata._compressed_klass; }
```

Comment:
Maybe we should rename this to e.g. `narrow_klass_legacy` to avoid accidentally using it? AFAIU, this is for legacy code paths?